packages/kbn-securitysolution-grouping/src/hooks/use_get_group_selector.tsx

```diff
@@ -112,33 +116,46 @@ export const useGetGroupSelector = ({
 
   const onChange = useCallback(
     (groupSelection: string) => {
+      let newSelectedGroups: string[] = [];
+      let sendTelemetry = true;
       // Simulate a toggle behavior when maxGroupingLevels is 1
       if (maxGroupingLevels === 1) {
-        setSelectedGroups([groupSelection]);
+        newSelectedGroups = [groupSelection];
       } else {
+        // if the group is already selected, remove it
         if (selectedGroups.find((selected) => selected === groupSelection)) {
+          sendTelemetry = false;
           const groups = selectedGroups.filter((selectedGroup) => selectedGroup !== groupSelection);
           if (groups.length === 0) {
-            setSelectedGroups(['none']);
+            newSelectedGroups = ['none'];
           } else {
-            setSelectedGroups(groups);
+            newSelectedGroups = groups;
           }
-          return;
+        } else {
```

Comment:
Most changes here were around removing the early return that was preventing onChange from triggering when a group is unselected but keeping the same behavior.